File: Searching_Algorithm/JAVA/BinarySearch_UsingRecursion.java
package Searching_Algorithm;
import java.util.*;
public class BinarySearch_UsingRecursion{
	public static int binarysearch(int arr[],int num,int si,int ei){
		if(si>ei){
			return -1;
		}
		int mid = si + (ei - si) / 2;
		if(arr[mid]==num){
			return mid;
		}
		else if(ei>mid){
			return binarysearch(arr,num,mid+1,ei);
		}
		else{
			return binarysearch(arr,num,si,mid-1);
		}
	}
    public static void main(String[]args){
		Scanner input=new Scanner(System.in);
		int[]arr={1,2,3,4,5,6,7,8};
		int k=6; 
		System.out.println(binarysearch(arr,k,0,arr.length-1));
		input.close();
	}
}

File: Searching_Algorithm/PYTHON/BinarySearch_UsingRecursion.py
def binarySearch(arr,num,start,end):
  if start > end:
    return -1
  mid = start + (end-start)//2
  if arr[mid] == num:
    return mid
  elif end > mid:
    return binarysearch(arr, num, mid + 1, end)
  else:
    return binarysearch(arr, num, start, mid - 1)

if __name__ == "__main__":
  n = int(input("Enter number of elements: "))
  array=[]
  print("Enter the elements:")
  for i in range(n):
    x = int(input(""))
    array.append(x)
  num = int(input("Enter the element to search: "))
  pos = binarySearch(array, num, 0, len(array)-1)
  if pos == -1:
    print("Element is not present in the array")
  else:
    print(f"Element found at position {pos} in the array")

  
